packages/apollo-gateway/src/executeQueryPlan.ts

```diff
@@ -229,11 +288,33 @@ async function executeFetch<TContext>(
     variables: Record<string, any>,
   ): Promise<ResultMap | void> {
     const source = print(operation);
+    // We declare this as 'any' because it is missing url and method, which
+    // GraphQLRequest.http is supposed to have if it exists.
+    let http: any;
+
+    // If we're capturing a trace for Engine, then save the operation text to
+    // the node we're building and tell the federated service to include a trace
+    // in its response.
+    if (traceNode) {
+      http = {
+        headers: new Headers({ 'apollo-federation-include-trace': 'ftv1' }),
```

Comment:
I think the hooking in happens at the GrapQLTransport level, so we might have some more work to do here, but I think this means that if users do that, they'll lose metrics tracking rather than losing their transport changes, which we should definitely resolve, but it's terrifying.